core/src/server/rpc_module.rs

```diff
@@ -842,6 +923,55 @@ impl Drop for PendingSubscription {
 	}
 }
 
+/// The result obtain from calling [`PendingSubscription::pipe_from_try_stream`] that
+/// can be utilized to execute specific operations depending on the result.
+#[derive(Debug)]
+pub enum PipeFromStreamResult {
+	/// The connection was accepted and the pipe returned [`SubscriptionClosed::Success`].
+	Success(SubscriptionSink),
+	/// The connection was accepted and the pipe returned [`SubscriptionClosed::Failed`]
+	/// with the provided error.
+	Failure(SubscriptionSink, ErrorObjectOwned),
+	/// The remote peer closed the connection or called the unsubscribe method.
+	RemotePeerAborted,
+}
+
+impl PipeFromStreamResult {
+	/// Callback that will run the provided function if the result is [`PipeFromStreamResult::Success`].
+	/// After the function runs a new [`PipeFromStreamResult::RemotePeerAborted`] is returned.
```

Comment:
Can you explain why `PipeFromStreamResult::RemotePeerAborted` is returned here?

Can't you borrow the `SubscriptionSink` in the closure?

Reply:
I was thinking of having a sort of `PipeFromStreamResult::ResultConsumed` to keep on chaining `on_success` with `on_failure` for better ergonomics. WDYT?

Reply:
sure, that works as well

Reply:
`ResultConsumed` makes sense but does mean you lose information if you're matching on the enum.

Another option might be something like:

```rust
pub enum PipeFromStreamResult {
	/// The connection was accepted and the pipe returned [`SubscriptionClosed::Success`].
	Success(Option<SubscriptionSink>),
	/// The connection was accepted and the pipe returned [`SubscriptionClosed::Failed`]
	/// with the provided error.
	Failure(Option<(SubscriptionSink, ErrorObjectOwned)>),
	/// The remote peer closed the connection or called the unsubscribe method.
	RemotePeerAborted,
}
```

And then `on_success` and `on_failure` only run if the options are `Some` and `take` the values out of them, but one can still match on the enum and see what the result was. (semantically it's the same as having `SuccessConsumed` and `FailureConsumed` variants, but less matching). I don't love it, but I haven't thought up anything better!

Reply:
The `Option` approach seems a bit better for the end user 👍

Reply:
d